doc/api/stream.md

```diff
@@ -2886,8 +2886,38 @@ readable.on('data', (chunk) => {
 
 #### Piping to writable streams from async iterators
 
-In the scenario of writing to a writable stream from an async iterator, ensure
-the correct handling of backpressure and errors.
+In the scenario of writing to a writable stream from an async iterator, the
+correct handling of backpressure and errors must be ensured.
+[`stream.pipeline()`][] conveniently abstracts away the handling of
+backpressure and backpressure-related errors:
+
+```js
+const { pipeline } = require('stream');
+const util = require('util');
+const fs = require('fs');
+
+const writable = fs.createWriteStream('./file');
+
+// Callback Pattern
+pipeline(iterator, writable, (err, value) => {
+  if (err) {
+    console.error(err);
+  } else {
+    console.log(value, 'value returned');
+  }
+});
+
+// Promise Pattern
+const pipelinePromise = util.promisify(pipeline);
+pipelinePromise(iterator, writable)
+  .then((value) => {
+    console.log(value, 'value returned');
+  })
+  .catch(console.error);
+```
+
+Although this is unnecessary if using [`stream.pipeline()`][], it is possible
+to manually handle backpressure and backpressure-related errors:
 
```

Comment:
I think we should remove the part below. Makes things complicated.

Reply:
I agree. I would not recommend folks to use it at all.

Reply:
Sounds good. I have removed that section completely.